lib/tooling/osaca-tool.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import fs from 'fs-extra';
+
+import { BaseTool } from './base-tool';
+
+export class OSACATool extends BaseTool {
+    static get key() { return 'osaca-tool'; }
+
+    rewriteAsm(asm) {
+        return asm;
+    }
+
+    writeAsmFile(data, destination) {
+        return fs.writeFile(destination, this.rewriteAsm(data));
+    }
+
+    async runTool(compilationInfo, inputFilepath, args) {
+        if (compilationInfo.filters.binary) {
+            return new Promise((resolve) => {
```

Comment:
this can be simplified to just:

```js
if (compilationInfo.filters.binary) {
    return this.createErrorResponse('<cannot run analysis on binary>');
}
```

Reply:
just noticed you copied these from existing code in x86to6502-tool - the difference is that your `runTool` is async, where the one in x86to6502 is not.

Reply:
Thanks for explaining the differences.